### ISSUE TITLE
When will springboot3.0.0 be supported 

### ISSUE THREAD

**Issue by clverhack** [apache/rocketmq-spring#505]
When will springboot3.0.0 be supported

**Comment by baopeizhou** [comment]
I download the source code of Rocket spring version 2.2.2; Based on this, I upgrade the Spring Boot dependency library version to 3.0.0 and the Spring dependency library version to 6.0.2, modify the automatic configuration file, repackage it, and tested it normal in the Spring Boot3. x application.

**Comment by baopeizhou** [comment]
1.Add the following dependencies
         <dependency>
            <groupId>jakarta.annotation</groupId>
            <artifactId>jakarta.annotation-api</artifactId>
        </dependency>
2.Add a new automation profile
![图片](https://user-images.githubusercontent.com/8037926/209072447-733bf512-0c3e-4cc6-90fa-8121f82f6b88.png)

**Comment by KouShenhai** [comment]
> 1.添加以下依赖项 jakarta.annotation jakarta.annotation-api 2.添加新的自动化配置文件 ![图片](https://user-images.githubusercontent.com/8037926/209072447-733bf512-0c3e-4cc6-90fa-8121f82f6b88.png)
rocketmq可以正常消费？

**Comment by dousp** [comment]
Many components do not provide adaptation now.
Upgrading to 3.0 by yourself is like changing a project. Various components in the project need to be adapted, and jar conflicts must be resolved. This feels like wearing a painful mask...

**Comment by baopeizhou** [comment]
> > 1.添加以下依赖项 jakarta.annotation jakarta.annotation-api 2.添加新的自动化配置文件 ![图片](https://user-images.githubusercontent.com/8037926/209072447-733bf512-0c3e-4cc6-90fa-8121f82f6b88.png)
> > Rocketmq can be consumed normally?
In my application, the test results are normal

**Comment by baopeizhou** [comment]
> Many components do not provide adaptation now. Upgrading to 3.0 by yourself is like changing a project. Various components in the project need to be adapted, and jar conflicts must be resolved. This feels like wearing a painful mask...

Yes, you are right; I found that the problem with the latest version is the incompatibility between annotations in JDK8 and JDK17, and the way of automatic configuration of state is different. In addition, the main dependency is spring basic components, so the customized upgrade version mainly corrects the annotations and upgrades the spring basic component version.

**Comment by KouShenhai** [comment]
Replace with the assembly method of spring boot 3
![image](https://user-images.githubusercontent.com/48756217/226252370-33b6bc6d-c5b0-488f-892c-9567b9747aa7.png)

**Comment by KouShenhai** [comment]
> 生效了嘛？没有报这个错误嘛？ 原因：org.springframework.beans.factory.BeanCreationException：创建名称为“org.apache.rocketmq.spring.support.DefaultRocketMQListenerContainer_1”的 bean 时出错：org/apache/rocketmq/common/protocol/heartbeat/MessageModel at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean（AbstractAutowireCapableBeanFactory.java：1770）在 org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean（AbstractAutowireCapableBeanFactory.java：598） 在 org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean（AbstractAutowireCapableBeanFactory.java：520） 在 org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0（AbstractBeanFactory.java：325）在 org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton（DefaultSingletonBeanRegistry.java：234） 在 org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean（AbstractBeanFactory.java：323） 在 org.springframework.beans.factory.support.AbstractBeanFactory.getBean（AbstractBeanFactory.java：204） 在 org.springframework.context.support.AbstractApplicationContext.getBean（AbstractApplicationContext.java：1161） at org.apache.rocketmq.spring.autoconfigure.ListenerContainerConfiguration.registerContainer（ListenerContainerConfiguration.java：102） at org.apache.rocketmq.spring.annotation.RocketMQMessageListenerBeanPostProcessor.postProcessAfterInitialization（RocketMQMessageListenerBeanPostProcessor.java：55）在 org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.applyBeanPostProcessorsAfterInitialization（AbstractAutowireCapableBeanFactory.java：434） 在 org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean（AbstractAutowireCapableBeanFactory.java：1773） 在 org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean（AbstractAutowireCapableBeanFactory.java：598） ...省略了 15 个常见帧 原因：java.lang.NoClassDefFoundError：org/apache/rocketmq/common/protocol/heartbeat/MessageModel at org.apache.rocketmq.spring.support.DefaultRocketMQListenerContainer.initRocketMQPushConsumer（DefaultRocketMQListenerContainer.java：656） at org.apache.rocketmq.spring.support.DefaultRocketMQListenerContainer.afterPropertiesSet（DefaultRocketMQListenerContainer.java：380）在 org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods（AbstractAutowireCapableBeanFactory.java：1817） 在 org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean（AbstractAutowireCapableBeanFactory.java：1766） ...省略了 27 个常见帧 原因：java.lang.ClassNotFoundException：org.apache.rocketmq.common.protocol.heartbeat.MessageModel at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass（BuiltinClassLoader.java：641） at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass（ClassLoaders.java：188） at java.base/java.lang.ClassLoader.loadClass（ClassLoader.java：525） ...省略了 31 个常见帧

平稳运行，没有问题，我已经升级spring boot 3.2.x

**Comment by KimRoyal22** [comment]
> > 生效了嘛？没有报这个错误嘛？ 原因：org.springframework.beans.factory.BeanCreationException：创建名称为“org.apache.rocketmq.spring.support.DefaultRocketMQListenerContainer_1”的 bean 时出错：org/apache/rocketmq/common/protocol/heartbeat/MessageModel at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean（AbstractAutowireCapableBeanFactory.java：1770）在 org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean（AbstractAutowireCapableBeanFactory.java：598） 在 org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean（AbstractAutowireCapableBeanFactory.java：520） 在 org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0（AbstractBeanFactory.java：325）在 org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton（DefaultSingletonBeanRegistry.java：234） 在 org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean（AbstractBeanFactory.java：323） 在 org.springframework.beans.factory.support.AbstractBeanFactory.getBean（AbstractBeanFactory.java：204） 在 org.springframework.context.support.AbstractApplicationContext.getBean（AbstractApplicationContext.java：1161） at org.apache.rocketmq.spring.autoconfigure.ListenerContainerConfiguration.registerContainer（ListenerContainerConfiguration.java：102） at org.apache.rocketmq.spring.annotation.RocketMQMessageListenerBeanPostProcessor.postProcessAfterInitialization（RocketMQMessageListenerBeanPostProcessor.java：55）在 org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.applyBeanPostProcessorsAfterInitialization（AbstractAutowireCapableBeanFactory.java：434） 在 org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean（AbstractAutowireCapableBeanFactory.java：1773） 在 org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean（AbstractAutowireCapableBeanFactory.java：598） ...省略了 15 个常见帧 原因：java.lang.NoClassDefFoundError：org/apache/rocketmq/common/protocol/heartbeat/MessageModel at org.apache.rocketmq.spring.support.DefaultRocketMQListenerContainer.initRocketMQPushConsumer（DefaultRocketMQListenerContainer.java：656） at org.apache.rocketmq.spring.support.DefaultRocketMQListenerContainer.afterPropertiesSet（DefaultRocketMQListenerContainer.java：380）在 org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods（AbstractAutowireCapableBeanFactory.java：1817） 在 org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean（AbstractAutowireCapableBeanFactory.java：1766） ...省略了 27 个常见帧 原因：java.lang.ClassNotFoundException：org.apache.rocketmq.common.protocol.heartbeat.MessageModel at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass（BuiltinClassLoader.java：641） at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass（ClassLoaders.java：188） at java.base/java.lang.ClassLoader.loadClass（ClassLoader.java：525） ...省略了 31 个常见帧
> 
> 平稳运行，没有问题，我已经升级spring boot 3.2.x

我的也报这个错，有没有解决办法呢
